src/main/java/org/hawkular/client/android/backend/BackendClient.java

```diff
@@ -211,6 +216,38 @@ public void getEnvironments(@NonNull Callback<List<Environment>> callback) {
         readPipe(BackendPipes.Names.ENVIRONMENTS, uri, callback);
     }
 
+
+    public void getFeeds(@NonNull Callback<List<Feed>> callback) {
+        URI uri = Uris.getUri(BackendPipes.Paths.FEEDS);
+
+        readPipe(BackendPipes.Names.FEEDS, uri, callback);
+    }
+
+    public void getResourcesFromFeed(@NonNull Callback<List<Resource>> callback, Feed feed) {
+        URI uri = Uris.getUri(String.format(BackendPipes.Paths.FEED_RESOURCES,
+                Uris.getEncodedParameter(feed.getId())));
+
+        readPipe(BackendPipes.Names.FEED_RESOURCES, uri, callback);
+    }
+
+
+    public void getRecResourcesFromFeed(@NonNull Callback<List<Resource>> callback, Resource resource) {
+
+            URI uri = Uris.getUri(String.format(BackendPipes.Paths.FEED_REC_RESOURCES,resource.getFeed(),
+                    resource.getId().replace("/","%2F")));
+
+        readPipe(BackendPipes.Names.FEED_REC_RESOURCES, uri, callback);
+    }
+
+
+    public void getMetricsFromFeed(@NonNull Callback<List<Metric>> callback, Resource resource) {
+        URI uri = Uris.getUri(String.format(BackendPipes.Paths.FEED_METRICS, resource.getFeed(),
+                resource.getId().replace("/","%2F")));
```

Comment:
This is not a proper way to encode the metricId. URLEncoder.encode() is the method you're searching for.